### PR TITLE
Bump to 1.28 k8s version

### PR DESCRIPTION
**Pull request by chajain**

### CODE REVIEW

**Comment by chajain** [comment]
@MbolotSuse The ci-build is failing with error

> go: downloading github.com/josharian/intern v1.0.0
> # k8s.io/client-go/applyconfigurations/meta/v1
> /go/pkg/mod/github.com/rancher/client-go@v1.27.4-rancher1/applyconfigurations/meta/v1/unstructured.go:64:38: cannot use doc (variable of type *"github.com/google/gnostic/openapiv2".Document) as *"github.com/google/gnostic-models/openapiv2".Document value in argument to proto.NewOpenAPIData
> time="2024-01-04T02:06:31Z" level=fatal msg="exit status 1"

I am guessing that I also need to update the client-go version in https://github.com/rancher/csp-adapter/pull/76/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L13 but the PR for that is still in WIP https://github.com/rancher/client-go/pull/7 .

**Review by guangyee — APPROVED** [review]
Looks good.

**Comment by MbolotSuse** [comment]
@chajain Looks like the CI is failing due to the adapter failing to build. You will need to update the client-go version as well. I would recommend putting this PR on hold until that work is complete.

File: go.mod [hunk 1]
```diff
@@ -5,28 +5,28 @@ go 1.20
...
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.27.4
-	k8s.io/apimachinery => k8s.io/apimachinery v0.27.4
-	k8s.io/apiserver => k8s.io/apiserver v0.27.4
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.27.4
+	k8s.io/api => k8s.io/api v0.28.4
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.28.4
+	k8s.io/apimachinery => k8s.io/apimachinery v0.28.4
+	k8s.io/apiserver => k8s.io/apiserver v0.28.4
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.28.4
 	k8s.io/client-go => github.com/rancher/client-go v1.27.4-rancher1
```
Comment (MbolotSuse) [inline go.mod]: Per conversation elsewhere, you'll need to bump this value as well.

**Review by MbolotSuse — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (guangyee) [reply go.mod]: updated

**Pull request merged**